### DISCUSSION TITLE
Element pivot 透视表

### DISCUSSION THREAD

**Discussion by jamie-mttk** [element-plus/element-plus#17208] [General]
A pivot table based on Element Plus Table
基于Element Plus表格的透视表插件

https://github.com/jamie-mttk/element-pivot/
<img width="1280" alt="screen" src="https://github.com/element-plus/element-plus/assets/123045766/5bb9918d-d2da-4c7f-b80f-1110adf5f123">

**Comment by warmthsea** [comment]
什么是透视表？它和平常的表格有啥区别吗？（从这个图上来看）

**Reply by jamie-mttk** [reply]
表格是从上往下显示，列固定；透视表是行和列动态显示，加差点显示数据。
如横向是省/市/区，纵向是产品分类/产品名称，交叉点显示销售数量和销售金额

**Reply by xing403** [reply]
多级表头？

**Reply by jamie-mttk** [reply]
一般element table 通过配置显示列的方式显示表格
透视表通过下面配置自动横向显示多级表头和竖向显示层级关系
具体代码参考  [https://github.com/jamie-mttk/element-pivot/blob/master/src/components/Demo.vue](https://github.com/jamie-mttk/element-pivot/blob/master/src/components/Demo.vue)

` <Pivot
    :data="data"
    :dimensionCol="dimensionCol"
    :dimensionRow="dimensionRow"
    :metric="metric"
    :option="{}"
  ></Pivot>`

各个参数配置(简化版)
`
const dimensionCol = [
  { key: 'province', label: 'Province' },
  { key: 'city', label: 'City' },
  { key: 'district', label: 'District' }
]

const dimensionRow = [
  { key: 'category', label: 'Category' },
  { key: 'goods', label: 'Goods', option: { widthPerChar: 10, widthBase: 32 } }
]

const metric = [  {    key: 'quantity',    label: 'Quantity',  },
  {    key: 'amount',    label: 'Amount',  }]`

**Comment by jinceon** [comment]
怎么没有投票选项。
我要点【反对】

**Reply by jamie-mttk** [reply]
为什么？

**Reply by jinceon** [reply]
小众需求自己写个库玩就行

**Comment by L0st1** [comment]
所以这个透视表属于是多维表格吗，类似于[这个](https://s2.antv.antgroup.com/zh/examples/basic/pivot/#grid)。感觉“透视表”这个表述并不方便理解

**Reply by jamie-mttk** [reply]
是的
下面是一个很好的透视表实现，功能很强
https://js.devexpress.com/Vue/Demos/WidgetsGallery/Demo/PivotGrid/Overview/MaterialBlueLight/
缺点 1. 虽然可以免费用但是是商业的，有法律风险  2.如果项目中使用了element plus，风格上很难统一

**Reply by L0st1** [reply]
实际上多维表格用合并行和合并列也可以完成，但是很费劲，合并行列如果类似于Grid布局就好使多了；
为什么不试试S2
